infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/fsm/InitializedState.java

```diff
@@ -77,6 +78,14 @@ public void onEntry(LogReplicationState from) {
 
     @Override
     public void onExit(LogReplicationState to) {
+        LogReplicationStateType type = to.getType();
+
+        if (type.equals(LogReplicationStateType.IN_SNAPSHOT_SYNC) || type.equals(LogReplicationStateType.WAIT_SNAPSHOT_APPLY)) {
+            fsm.getAckReader().setSyncType(SyncType.SNAPSHOT);
```

Comment:
my previous comment/confusion was mostly introduced by this in-memory `lastSyncType` as well. Actually in polling task it will never be null, since it's reading from statusTable which already initialized. But the no-op is also there by checking `SyncStatus.NOT_STARTED`

Reply:
I see your point, Chris.  Basically, the polling task need not keep track of the ongoing sync type.  It can rely on the whatever is already written to the table.  Let us discuss this and finalize.